cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/operationrunner/InMemorySourceControlOperationRunner.java

```diff
@@ -49,28 +46,32 @@ public class InMemorySourceControlOperationRunner implements SourceControlOperat
   private static final Logger LOG = LoggerFactory.getLogger(InMemorySourceControlOperationRunner.class);
 
   @Inject
-  public InMemorySourceControlOperationRunner(RepositoryManagerFactory repoManagerFactory) {
+  InMemorySourceControlOperationRunner(RepositoryManagerFactory repoManagerFactory) {
     this.repoManagerFactory = repoManagerFactory;
   }
 
   @Override
   public PushAppResponse push(NamespaceId namespace, RepositoryConfig repoConfig,
                               ApplicationDetail appToPush, CommitMeta commitDetails)
     throws PushFailureException, NoChangesToPushException, AuthenticationConfigException {
-    try (RepositoryManager repositoryManager = repoManagerFactory.create(namespace, repoConfig)) {
+    RepositoryManager repositoryManager = repoManagerFactory.create(namespace, repoConfig);
+    try {
       try {
         repositoryManager.cloneRemote();
-      } catch (GitAPIException e) {
-        throw new PushFailureException("Failed to clone remote repository", e);
+      } catch (GitAPIException | IOException e) {
+        throw new PushFailureException(String.format("Failed to clone remote repository: %s", e.getMessage()));
```

Comment:
should always include the cause in an exception so the full context is not lost:

`throw new PushFailureException(String.format("...", e.getMessage()), e)`

Reply:
Added back the cause exception in throw